src/com/liakhovskyi/LinkedList.java

```diff
@@ -80,6 +75,26 @@ public String print() {
         return stringBuilder.toString();
     }
 
+    private void checkForIndexException(int index) {
+        if (index >= size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+    }
+
+    private void checkForEmptyListException() {
```

Comment:
this is not a correct check
delete this method
if you have empty list and try get or remove some element your "checkForIndexException" will throw a correct exception